modules/javafx.web/src/test/java/test/javafx/scene/web/LocalStorageTest.java

```diff
@@ -0,0 +1,153 @@
+import javafx.concurrent.Worker;
+import javafx.scene.web.WebView;
+import javafx.scene.web.WebEngine;
+
+
+public class LocalStorageTest extends TestBase {
+
+    private static final File LOCAL_STORAGE_DIR = new File("LocalStorageDir");
+    private static final File PRE_LOCKED = new File("zoo_local_storage");
+
+    private static RandomAccessFile preLockedRaf;
+    private static FileLock preLockedLock;
+    private static final Random random = new Random();
+
+    private static void deleteRecursively(File file) throws IOException {
+        if (file.isDirectory()) {
+            for (File f : file.listFiles()) {
+                deleteRecursively(f);
+            }
+        }
+        if (!file.delete()) {
+            // If WebKit takes time to close the file, better
+            // delete it during VM shutdown.
+            file.deleteOnExit();
+        }
+    }
+
+    private WebEngine createWebEngine() {
+        return submit(() -> new WebEngine());
+    }
+
+    /* test localstorage instance */
+    void checkLocalStorageAfterWindowClose(WebEngine webEngine) {
+        load(new File("src/test/resources/test/html/localstorage.html"));
+        submit(() -> {
+            assertNotNull(webEngine.executeScript("localStorage;"));
+            getEngine().executeScript("window.close();");
+            assertNotNull(webEngine.executeScript("localStorage;"));
+        });
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws IOException {
+        PRE_LOCKED.mkdirs();
+        File preLockedFile = new File(PRE_LOCKED, ".lock");
+        preLockedRaf = new RandomAccessFile(preLockedFile, "rw");
+        preLockedLock = preLockedRaf.getChannel().tryLock();
+        if (preLockedLock == null) {
+            fail(format("Directory [%s] is already locked "
+                    + "externally", PRE_LOCKED));
+        }
+    }
+
+    @AfterClass
+    public static void afterClass() throws IOException {
+        preLockedLock.release();
+        preLockedRaf.close();
+        deleteRecursively(LOCAL_STORAGE_DIR);
+        deleteRecursively(PRE_LOCKED);
+    }
+
+    @Test
+    public void testLocalStorage() throws Exception {
+        final WebEngine webEngine = getEngine();
+        webEngine.setJavaScriptEnabled(true);
+        webEngine.setUserDataDirectory(LOCAL_STORAGE_DIR);
+        checkLocalStorageAfterWindowClose(webEngine);
+    }
+
+    /* test localstorage set data before window.close and check data after window.close */
+    @Test
+    public void testLocalStorageData() {
+        load(new File("src/test/resources/test/html/localstorage.html"));
+        submit(() -> {
+            WebView view = getView();
+            view.getEngine().executeScript("test_local_storage_set();"); // set data
+            getEngine().executeScript("window.close();");
+            //get data
+            String s = (String) view.getEngine().executeScript("document.getElementById('key').innerText;");
+            assertEquals(s, "1001");
```

Comment:
The arguments should be switched (expected comes first).

Reply:
Agree , done